### PR TITLE
Mk/fix multi sheet

### PR DESCRIPTION
**Pull request by mkangia**
[Ticket](https://github.com/dimagi/commcare-hq/pull/25365)
Currently when a language is selected on App Translations, we assume that it's a single sheet.
That does not work well with validating translations where its possible for a user to select a language but ask for validation.

This PR makes single sheet expectation based on the use case.
And then modifies validation logic to support multi sheet with language selection possible and also makes it mandatory to select language when validating

Also fixing an exception that occurs when module/form in the sheet is not present in the application in [commit](https://github.com/dimagi/commcare-hq/commit/de523bf7e365a0961583b93320384a5d86283ece)

FF: `VALIDATE_APP_TRANSLATIONS`

### CODE REVIEW

File: corehq/apps/translations/app_translations/upload_app.py [hunk 1]
```diff
@@ -96,9 +96,12 @@ def get_expected_headers(sheet_name):
     if error:
         msgs.append((messages.error, error))
         return msgs
-
-    expected_headers_by_sheet_name = {k: v for k, v in get_bulk_app_sheet_headers(app, lang=lang)}
-    expected_headers_by_id = {k: v for k, v in get_bulk_app_sheet_headers(app, lang=lang, by_id=True)}
+    # if there is a lang selected, assume that user submitted a single sheet
+    single_sheet = bool(lang)
```
Comment (orangejenny) [inline corehq/apps/translations/app_translations/upload_app.py]: There's an `is_single_sheet` function based on sheet name used down on line 118/121, I'd use that here.

File: corehq/apps/translations/app_translations/utils.py [hunk 2]
```diff
@@ -12,7 +12,8 @@
 from corehq.apps.translations.generators import EligibleForTransifexChecker
 
 
-def get_bulk_app_sheet_headers(app, lang=None, eligible_for_transifex_only=False, by_id=False):
+def get_bulk_app_sheet_headers(app, single_sheet=False, lang=None, eligible_for_transifex_only=False,
```
Comment (orangejenny) [inline corehq/apps/translations/app_translations/utils.py]: `lang` is no longer used.

File: corehq/apps/translations/views.py [hunk 3]
```diff
@@ -95,7 +95,10 @@ def download_bulk_app_translations(request, domain, app_id):
     lang = request.GET.get('lang')
     skip_blacklisted = request.GET.get('skipbl', 'false') == 'true'
     app = get_app(domain, app_id)
-    headers = get_bulk_app_sheet_headers(app, lang=lang, eligible_for_transifex_only=skip_blacklisted)
+    # if there is a lang selected, assume that user wants a single sheet
+    single_sheet = bool(lang)
```
Comment (orangejenny) [inline corehq/apps/translations/views.py]: Now that `single_sheet` exists, it'd be more readable to also use it in the `if lang` check below.

**Review by orangejenny — APPROVED** [review]
Made a couple of cleanup suggestions, nothing blocking if this needs to go out soon.

File: corehq/apps/translations/app_translations/utils.py (hunk 2 above)
Comment (mkangia) [reply corehq/apps/translations/app_translations/utils.py]: `langs = [lang] if single_sheet`

File: corehq/apps/translations/app_translations/upload_app.py (hunk 1 above)
Comment (mkangia) [reply corehq/apps/translations/app_translations/upload_app.py]: That happens within the sheets iteration, i would need to do something like 
`is_single_sheet(workbook.worksheets[0].title)` here which sounds fine for now but brittle.

**Review by kaapstorm — APPROVED** [review]

**Pull request merged**